nilearn/glm/first_level/first_level.py

```diff
@@ -1060,6 +1034,249 @@ def first_level_from_bids(dataset_path, task_label, space_label=None,
                                  (len(events), len(imgs)))
             confounds = [pd.read_csv(c, sep='\t', index_col=None)
                          for c in confounds]
+
             models_confounds.append(confounds)
 
     return models, models_run_imgs, models_events, models_confounds
+
+
+def _bids_validate_arguments(dataset_path: str,
+                             task_label: str,
+                             space_label: str | None,
+                             img_filters: list[Tuple[str, str]],
+                             derivatives_path: str,
+                             supported_filters: list[str],
+                             ) -> None:
+    """Check type and value of arguments of first_level_from_bids.
+
+    Parameters
+    ----------
+    dataset_path : :obj:`str`
+        _description_
+    task_label : :obj:`str`
+        _description_
+    space_label : :obj:`str` or None
+        _description_
+    img_filters : list[Tuple[str, str]]
+        _description_
+    derivatives_path : :obj:`str`
+        _description_
+    supported_filters : list[str]
+        _description_
+
+    Raises
+    ------
+    TypeError
+        _description_
+    ValueError
+        _description_
+    """    
+    if not isinstance(dataset_path, str):
+        raise TypeError(
+            "dataset_path must be a string, "
+            f"instead {type(dataset_path)} was given"
+        )
+    if not os.path.exists(dataset_path):
+        raise ValueError(f'given path do not exist: {dataset_path}')
+
+    if not os.path.exists(derivatives_path):
+        raise ValueError('derivatives folder does not exist in given dataset')
+
+    if not isinstance(task_label, str):
+        raise TypeError(
+            "task_label must be a string, "
+            f"instead {type(task_label)} was given"
+        )
+
+    if space_label is not None and not isinstance(space_label, str):
+        raise TypeError(
+            "space_label must be a string, "
+            f"instead {type(space_label)} was given."
+        )
+
+    if not isinstance(img_filters, list):
+        raise TypeError(
+            f"img_filters must be a list, "
+            f"instead {type(img_filters)} was given."
+        )
+    for this_filter in img_filters:
+        if (not isinstance(this_filter[0], str)
+                or not isinstance(this_filter[1], str)):
+            raise TypeError('filters in img filters must be (str, str), '
+                            f"instead {type(this_filter)} was given.")
+        if this_filter[0] not in supported_filters:
+            raise ValueError(
+                f"field {this_filter[0]} is not a possible filter. "
+                f"Only {supported_filters} are allowed.")
+
+
+def _bids_filter(task_label: str,
```

Comment:
I may choose another name because this returns a "filter" and calling it `_filter_files_from_bids_dataset` may be read as it actually performs some filtering action.